src/impl/xhr.js

```diff
@@ -1,15 +1,21 @@
-import { uint8ArrayFromString } from '../util';
+import { root, TextEncoderPolyfill } from '../util';
 
 export const XHR = 'xhr';
 
 export default function xhrRequest(options) {
+  let textEncoder;
+  if (typeof root.TextEncoder !== 'undefined') {
```

Comment:
This could be improved... to this check in util.js or add TextEncoder to global/root.